src/Compilers/CSharp/Portable/Parser/DirectiveParser.cs

```diff
@@ -461,64 +461,86 @@ private DirectiveTriviaSyntax ParsePragmaDirective(SyntaxToken hash, SyntaxToken
             {
                 var warning = this.EatContextualToken(SyntaxKind.WarningKeyword);
                 SyntaxToken style;
-                if (this.CurrentToken.Kind == SyntaxKind.DisableKeyword || this.CurrentToken.Kind == SyntaxKind.RestoreKeyword)
+                if (this.CurrentToken.Kind == SyntaxKind.DisableKeyword || this.CurrentToken.Kind == SyntaxKind.RestoreKeyword ||
```

Comment:
>if [](start = 16, length = 2)

Consider using `switch (this.CurrentToken.Kind)`.

Reply:
> Consider using switch (this.CurrentToken.Kind).

I do not think the use of switch is warranted here. If you are concerned about multiple evaluations of ```this.CurrentToken.Kind``` property, I can cache its value in a local and reuse. Let me know.

---
In reply to: [245823259](https://github.com/dotnet/roslyn/pull/32208#discussion_r245823259) [](ancestors = 245823259)